packages/data-service/src/csfle-collection-tracker.ts

```diff
@@ -437,53 +446,70 @@ export class CSFLECollectionTrackerImpl implements CSFLECollectionTracker {
     filter: Document,
     collectionInfos: CollectionInfo[]
   ): Error | undefined {
-    if (typeof filter?.name !== 'string' || collectionInfos.length > 1) {
+    // filter is either { name: string } or { name: { $in: string[] } }
```

Comment:
can this be documented as a type?

Reply:
Yeah, but the point here is that the input isn't of a known type ... we could split this out into a separate validation function if you prefer, that takes a generic `unknown` or `Document` input and returns this type

Reply:
I see, I don't have strong feelings about it; leaving it as is is fine also 👍